src/express.re

```diff
@@ -268,7 +268,10 @@ module App = {
   external asMiddleware : t => Middleware.t = "%identity";
   /** [asMiddleware app] casts an App instance to a Middleware type */
 
-  external listen : t => port::int => unit = "" [@@bs.send];
+  external listen_ : t => port::int => onListen::(Js.Null_undefined.t Js.Exn.t => unit) [@bs.uncurry] => unit = "" [@@bs.send];
+
+  let listen app ::port=3000 ::onListen=(fun _ => ()) => listen_ app ::port ::onListen;
```

Comment:
This way, listen can be called like so, where port and onListen are both optional and default to 3000 and a noop

```reason
listen app ();
listen app port::1000 ();
listen app onListen::(fun x => Js.log x) ();
listen app port::1000 onListen::(fun x => Js.log x) ();
```